zerver/views/messages.py

```diff
@@ -1252,6 +1253,9 @@ def send_message_backend(request: HttpRequest, user_profile: UserProfile,
             return json_error(_("Missing sender"))
         if message_type_name != "private" and not is_super_user:
             return json_error(_("User not authorized for this query"))
+        # For now, mirroring only works with recipient emails, not for
+        # recipient user IDs.
+        message_to = cast(Sequence[str], message_to)
```

Comment:
This should be enforced by an actual check, not just a typing system cast.

Reply:
e.g. `if not all(isinstance(to_item, str) for to_item in message_to):`

Reply:
Okay, so I added the actual check that enforces this, but we still need the manual cast after that check so that `mypy` doesn't complain about `create_mirrored_send_message` not being able to accept a `Sequence[int]` type parameter.